topics/tutorials/code_inspections.md

```diff
@@ -68,55 +66,32 @@ One of these classes is a good basis for a new inspection implementation, but a
-* `visitBinaryExpression()`, which does all the heavy lifting.
-  It is called to evaluate a `PsiBinaryExpression`, and it checks to see if the operands are `==` or `!=`, and if the operands are classes relevant to this inspection.
-* `isCheckedType()` evaluates the `PsiType` of the operands to determine if they are of interest to this inspection.
+The `ComparingStringReferencesInspection.buildVisitor()` method creates an anonymous visitor class based on [`JavaElementVisitor`](%gh-ic%/java/java-psi-api/src/com/intellij/psi/JavaElementVisitor.java) to traverse the PSI tree of the Java file being edited, inspecting for suspect syntax.
+The anonymous class overrides `visitBinaryExpression()`, which checks if a `PsiBinaryExpression`'s operator is `==` or `!=`, and if both operand types are `String`.
 
 ### Quick Fix Implementation
 
 The quick fix class acts much like an intention, allowing the user to invoke it on the `PsiElement` (or `TextRange`) highlighted by the inspection.
 
-The `ComparingReferencesInspection` implementation uses the nested class `CriQuickFix` to implement a quick fix based on [`LocalQuickFix`](%gh-ic%/platform/analysis-api/src/com/intellij/codeInspection/LocalQuickFix.java).
-The `CriQuickFix` class gives a user the option to change the use of `a == b` and `a != b` expression to `a.equals(b)` and `!a.equals(b)` respectively.
+The `ComparingStringReferencesInspection` implementation uses the nested class `ReplaceWithEqualsQuickFix` to implement a quick fix based on [`LocalQuickFix`](%gh-ic%/platform/analysis-api/src/com/intellij/codeInspection/LocalQuickFix.java).
+The `ReplaceWithEqualsQuickFix` class allows the user to change the use of `a == b` and `a != b` expression to `a.equals(b)` and `!a.equals(b)` respectively.
 
-The heavy lifting is done in `CriQuickFix.applyFix()`, which manipulates the PSI tree to convert the expressions.
+The heavy lifting is done in `ReplaceWithEqualsQuickFix.applyFix()`, which manipulates the PSI tree to convert the expressions.
 The change to the PSI tree is accomplished by the usual approach to modification:
 * Getting a `PsiElementFactory`.
 * Creating a new `PsiMethodCallExpression`.
 * Substituting the original left and right operands into the new `PsiMethodCallExpression`.
 * Replacing the original binary expression with the `PsiMethodCallExpression`.
 
-### Inspection Preferences Panel
-
-The inspection preferences panel is used to display information and provide additional options for the inspection.
-
-The panel created by `ComparingReferencesInspection.createOptionsPanel()` just defines a single `JTextField` to display in a `JPanel`.
-This `JPanel` gets added to the <control>Inspections</control> settings dialog when the inspection is selected.
-The `JTextField` allows editing of the `CHECKED_CLASSES` field while displayed in the panel.
-
-For simple customization requirements, see also:
```

Comment:
please don't remove this section completely. move it to new settings page "old" approach section

Reply:
Good catch, thanks. I didn't notice these are actual panels. I thought these are just example inspection implementations.
I moved them to the new page.